### PR TITLE
Properties

### PR DESCRIPTION
**Pull request by sepandhaghighi**
#### Reference Issues/PRs
#84 
#### What does this implement/fix? Explain your changes.
- `is_weighted` function added
- `_write_properties_to_json` function added
- `properties` section added to JSON, YAML and Pickle formats
- `_write_to_json` function renamed to `_write_data_to_json`

### CODE REVIEW

**Comment by sepandhaghighi** [comment]
@sadrasabouri Please review this PR 🥇

File: pyrgg/functions.py [hunk 1]
```diff
@@ -23,6 +23,25 @@
...
+    :type max_weight: int
+    :param min_weight: minimum weight
+    :type min_weight: int
+    :param signed: weight sign flag
+    :type signed: bool
+    :return: result as bool
+    """
+    if max_weight == min_weight and min_weight == 0:
+        return False
+    if max_weight == min_weight and min_weight == 1 and not signed:
```
Comment (sadrasabouri) [inline pyrgg/functions.py:40]: I can't figure out the relation between not being signed and not begin weighted, here.

**Review by sadrasabouri — APPROVED** [review]
Can you please add the reference which we talked about so that our definition for weighted graph will be well-accepted?

File: pyrgg/functions.py (hunk 1 above)
Comment (sepandhaghighi) [reply pyrgg/functions.py:40]: Consider a graph with `1` and `-1` weights.

Comment (sadrasabouri) [reply pyrgg/functions.py:40]: Got it :+1:

**Comment by sepandhaghighi** [comment]
> Can you please add the reference which we talked about so that our definition for weighted graph will be well-accepted?

Good point 💯 
I will update the `Reference` section in another PR

**Pull request merged**